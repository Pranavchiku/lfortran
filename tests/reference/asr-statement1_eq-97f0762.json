{
    "basename": "asr-statement1_eq-97f0762",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/statement1_eq.f90",
    "infile_hash": "d3bd808a7618b699049cb041c65b28c6359fef84ea952fdb8644b3b2",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-statement1_eq-97f0762.stdout",
    "stdout_hash": "08dead0e7cffe7b68002c09b76ca5b1ac7203e1991bbf04a6f7c99e9",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}