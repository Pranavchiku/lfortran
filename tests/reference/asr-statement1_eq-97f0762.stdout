(TranslationUnit 
    (SymbolTable
        1
        {
            dfloat:
                (Function 
                    (SymbolTable
                        11
                        {
                            dfloat:
                                (Variable 
                                    11 
                                    dfloat 
                                    ReturnVar 
                                    () 
                                    () 
                                    Default 
                                    (Real 8 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            i:
                                (Variable 
                                    11 
                                    i 
                                    In 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            j:
                                (Variable 
                                    11 
                                    j 
                                    In 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    dfloat 
                    [(Var 11 i)
                    (Var 11 j)] 
                    [(= 
                        (Var 11 dfloat) 
                        (Cast 
                            (IntegerBinOp 
                                (Var 11 i) 
                                Mul 
                                (Var 11 j) 
                                (Integer 4 []) 
                                ()
                            ) 
                            IntegerToReal 
                            (Real 8 []) 
                            ()
                        ) 
                        ()
                    )] 
                    (Var 11 dfloat) 
                    Source 
                    Public 
                    Implementation 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false. 
                    .false. 
                    [] 
                    [] 
                    .false.
                ), 
            dp:
                (Variable 
                    1 
                    dp 
                    Local 
                    (FunctionCall 
                        1 kind 
                        () 
                        [((RealConstant 
                            0.000000 
                            (Real 8 [])
                        ))] 
                        (Integer 4 []) 
                        (IntegerConstant 8 (Integer 4 [])) 
                        ()
                    ) 
                    (IntegerConstant 8 (Integer 4 [])) 
                    Parameter 
                    (Integer 4 []) 
                    Source 
                    Public 
                    Required 
                    .false.
                ), 
            kind:
                (ExternalSymbol 
                    1 
                    kind 
                    3 kind 
                    lfortran_intrinsic_kind 
                    [] 
                    kind 
                    Private
                ), 
            lfortran_intrinsic_kind:
                (IntrinsicModule lfortran_intrinsic_kind), 
            temp:
                (Function 
                    (SymbolTable
                        12
                        {
                            i:
                                (Variable 
                                    12 
                                    i 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            temp:
                                (Variable 
                                    12 
                                    temp 
                                    ReturnVar 
                                    () 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                )
                            
                        }) 
                    temp 
                    [] 
                    [(= 
                        (Var 12 i) 
                        (Cast 
                            (RealBinOp 
                                (FunctionCall 
                                    1 dfloat 
                                    () 
                                    [((IntegerConstant 1 (Integer 4 [])))
                                    ((IntegerConstant 2 (Integer 4 [])))] 
                                    (Real 8 []) 
                                    () 
                                    ()
                                ) 
                                Mul 
                                (Cast 
                                    (IntegerConstant 9 (Integer 4 [])) 
                                    IntegerToReal 
                                    (Real 8 []) 
                                    (RealConstant 
                                        9.000000 
                                        (Real 8 [])
                                    )
                                ) 
                                (Real 8 []) 
                                ()
                            ) 
                            RealToInteger 
                            (Integer 4 []) 
                            ()
                        ) 
                        ()
                    )] 
                    (Var 12 temp) 
                    Source 
                    Public 
                    Implementation 
                    () 
                    .false. 
                    .false. 
                    .false. 
                    .false. 
                    .false. 
                    [] 
                    [] 
                    .false.
                )
            
        }) 
    []
)
